action.yml

```diff
@@ -0,0 +1,199 @@
+          exit 1
+        fi
+
+        # Download composer.json for use in branches.php
+        curl -s -o __composer.json https://raw.githubusercontent.com/$GITHUB_REPOSITORY/$DEFAULT_BRANCH/composer.json
+
+        BRANCHES=$(MINIMUM_CMS_MAJOR=$MINIMUM_CMS_MAJOR DEFAULT_BRANCH=$DEFAULT_BRANCH php ${{ github.action_path }}/branches.php)
+        echo "BRANCHES is $BRANCHES"
+        if [[ $BRANCHES =~ "^FAILURE \- (.+)$" ]]; then
+          MESSAGE=${BASH_REMATCH[1]}
+          echo "Exception in branches.php - $MESSAGE"
+          exit 1
+        fi
+        if [[ $BRANCHES == "" ]]; then
+          echo "No branches to merge-up"
+          exit 0
+        fi
+        echo "branches=$BRANCHES" >> $GITHUB_OUTPUT
+        rm __tags.json
+        rm __branches.json
+        rm __composer.json
+
+        # Check to see if there is anything to merge-up
+        # Technically we could skip these API calls and just do the checking using only git, though
+        # doing this way not checkout the entire git history of the repo if possible
+        # These API calls are fast so it really doesn't add much overhead
+        # Downside to this is that we will abort early and not merge-up anything when we may have been
+        # able to say merge-up 4.13 -> 4 but not 4 -> 5.0
+        FROM_BRANCH=""
+        INTO_BRANCH=""
+        for BRANCH in $BRANCHES; do
+          FROM_BRANCH=$INTO_BRANCH
+          INTO_BRANCH=$BRANCH
+          if [[ $FROM_BRANCH == "" ]]; then
+            continue
+          fi
+          # https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits
+          RESP_CODE=$(curl -w %{http_code} -s -o __compare.json \
+          -X GET "https://api.github.com/repos/$GITHUB_REPOSITORY/compare/$INTO_BRANCH...$FROM_BRANCH" \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ github.token }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          )
+          if [[ $RESP_CODE != "200" ]]; then
+            echo "Unable to compare branches - HTTP response code was $RESP_CODE"
+            exit 1
+          fi
+          FILES=$(jq -r .files[].filename __compare.json)
+          rm __compare.json
+          
+          # Don't allow merge-ups when there are changes in dependency files
+          DEPENDENCY_FILES="composer.json package.json yarn.lock"
+          for DEPENDENCY_FILE in $DEPENDENCY_FILES; do
+            if [[ $(echo "$FILES" | grep $DEPENDENCY_FILE) != "" ]]; then
+              echo "Unable to mergeup between $FROM_BRANCH and $INTO_BRANCH - there are changes in $DEPENDENCY_FILE"
+              exit 1
+            fi
+          done
+
+          # Don't allow merge-ups when there are JS changes that would require a yarn build
+          if [[ $(echo "$FILES" | grep client/) != "" ]]; then
+            echo "Unable to mergeup between $FROM_BRANCH and $INTO_BRANCH - there are changes to JS files"
+            exit 1
+          fi
```

Comment:
I don't think we should do this. If there's conflicts, it will fail on a conflict. If there's no conflicts, then there's no problem. I believe that's the consensus we reached during backlog refinement as well, which is why this wasn't added as an acceptance criterion.

Even if we did do this, we should only do it on `client/dist`.

```suggestion
```

Reply:
I'm thinking we should keep this here for now even though simply because we have a [separate card](https://github.com/silverstripe/.github/issues/77) to address JS, so that seems like a better place to deal with this block

Reply:
I really disagree here. We explicitly discussed this in backlog refinement and decided to let these just fail due to merge conflicts. Again, if there's no conflict then there's no problem.

If you do feel strongly about keeping this, please at least only check for changes in `client/dist/` explicitly,

Reply:
OK have changed to `client/dist`